### DISCUSSION TITLE
Why not factor out the the elisp functions of apps?

### DISCUSSION THREAD

**Discussion by c1-g** [emacs-eaf/emacs-application-framework#712] [Q&A] [answered]
Right now eaf.el has about 2900+ lines of code. I anticipate that this will increase in the future and be harder to keep track of things. Furthermore, most of them are defined as autoloads anyway so they can be separated into their own file for example, the 8 functions and 5 variables related to the pdf-viewer app should be put in their own .el file in something like `app/pdf-viewer/eaf-pdf-viewer.el` so users can pick-and-choose which apps they want to use by simply loading them as library in their init file, developers will have an easier time changing things for the apps and maintainers will be able to do their job better.  Of course, we should leave the epc "backend" stuff in eaf.el and some essentials like `eaf-open`. 

Is there something preventing this?

**Comment by manateelazycat** [comment, marked as answer]
Done, thanks for suggestion!
